spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/VisionPredictor.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.optim
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.dataset.{Sample, SampleToMiniBatch}
+import com.intel.analytics.bigdl.models.utils.ModelBroadcast
+import com.intel.analytics.bigdl.nn.SpatialShareConvolution
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.transform.vision.image.{DistributedImageFrame, ImageFeature, ImageFrame}
+
+import scala.reflect.ClassTag
+
+/**
+ * VisionPredictor is a predictor for images
+ *
+ * @param model prediction model
+ * @param postProcessor post process after model forward
+ */
+class VisionPredictor[T: ClassTag](
+  model: Module[T],
```

Comment:
Do we need to preprocessor ?  As I read from `VisionPredictorSpec`, the way how we use this API seems a bit complicated, Maybe either applying a `preprocessor` in the `predict` life cycle or provide a method to specify transformer inside `ImageFrame` as below code is not that readable from user's perspective

```scala
 val imageFrame = ImageFrame.read(resource.getFile, sc) ->
    Resize(256, 256) -> CenterCrop(224, 224) ->
    ChannelNormalize(0.485f, 0.456f, 0.406f, 0.229f, 0.224f, 0.225f) -> MatToFloats() ->
    FloatsToTensor() -> ImageFrameToSample()
```